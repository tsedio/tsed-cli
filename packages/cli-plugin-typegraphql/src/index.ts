import "./templates/index.js";

import {TypeGraphqlModule} from "./TypeGraphqlModule.js";

export * from "./hooks/TypeGraphqlInitHook.js";

export default TypeGraphqlModule;
