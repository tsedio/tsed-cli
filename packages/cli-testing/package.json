{
  "name": "@tsed/cli-testing",
  "description": "Utils to test you CLI based on Ts.ED CLI",
  "version": "6.4.4",
  "type": "module",
  "main": "./lib/esm/index.js",
  "source": "./src/index.ts",
  "module": "./lib/esm/index.js",
  "typings": "./lib/types/index.d.ts",
  "exports": {
    ".": {
      "cli-tsed-source": "./src/index.ts",
      "types": "./lib/types/index.d.ts",
      "import": "./lib/esm/index.js",
      "default": "./lib/esm/index.js"
    }
  },
  "scripts": {
    "build": "yarn build:ts",
    "build:ts": "tsc --build tsconfig.esm.json"
  },
  "keywords": [
    "Ts.ED",
    "cli",
    "typescript",
    "framework",
    "rest",
    "api",
    "express",
    "decorators"
  ],
  "dependencies": {
    "@tsed/cli-core": "workspace:*",
    "tslib": "2.7.0"
  },
  "devDependencies": {
    "@tsed/typescript": "workspace:*",
    "cross-env": "7.0.3",
    "typescript": "5.6.2",
    "vitest": "3.2.4"
  }
}
