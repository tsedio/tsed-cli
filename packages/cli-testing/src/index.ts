export * from "./CliPlatformTest.js";
export * from "./FakeCliExeca.js";
export * from "./FakeCliFs.js";
export * from "./FakeCliHttpClient.js";
export * from "./normalizePath.js";
