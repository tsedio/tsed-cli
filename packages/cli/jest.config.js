// For a detailed explanation regarding each configuration property, visit:
// https://jestjs.io/docs/en/configuration.html

module.exports = {
  ...require("@tsed/jest-config"),
  roots: ["<rootDir>/src", "<rootDir>/test"],

  coverageThreshold: {
    global: {
      statements: 93.01,
      branches: 80.73,
      functions: 83.47,
      lines: 93.01
    }
  }
};
