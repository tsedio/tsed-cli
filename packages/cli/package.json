{
  "name": "@tsed/cli",
  "description": "CLI to bootstrap your Ts.ED project",
  "version": "6.1.10-rc.2",
  "type": "module",
  "main": "./lib/esm/index.js",
  "source": "./src/index.ts",
  "module": "./lib/esm/index.js",
  "typings": "./lib/types/index.d.ts",
  "exports": {
    ".": {
      "types": "./lib/types/index.d.ts",
      "import": "./lib/esm/index.js",
      "default": "./lib/esm/index.js"
    }
  },
  "scripts": {
    "build": "yarn build:ts",
    "build:ts": "tsc --build tsconfig.json",
    "run-cmd": "cross-env NODE_ENV=development CLI_MODE=ts CI=true node --import @swc-node/register/esm-register src/bin/tsed.ts",
    "start:help": "yarn run-cmd -h",
    "start:help:g": "yarn run-cmd generate -h",
    "start:help:i": "yarn run-cmd init -h",
    "start:version": "yarn run-cmd --version",
    "start:init:help": "yarn run-cmd init -h",
    "start:init:test": "yarn run-cmd init init -r ./.tmp/init/default --features=oidc --arch=default --convention=conv_default --platform=express --package-manager=npm --skip-prompt .",
    "start:init:test:jest": "yarn run-cmd init -r ./.tmp/init/default --features=jest --arch=default --convention=conv_default --platform=express --package-manager=npm --skip-prompt .",
    "start:init:run": "yarn run-cmd init -r ./.tmp/init/default",
    "start:init:params": "yarn run-cmd init -r ./.tmp/init/default --skip-prompt --features swagger,jest,lintstaged",
    "start:init:run:name": "yarn run-cmd init -r ./.tmp/init awesome --verbose",
    "start:init:args": "yarn run-cmd init -r ./.tmp/cmd --package-manager=npm --project-name=untitled3 --features=graphql,typeorm,passportjs,socketio,linter,eslint,prettier --arch=default --convention=default --platform=express --skip-prompt .",
    "start:add:run": "yarn run-cmd add -r ./.tmp",
    "start:generate": "yarn run-cmd generate -r ./.tmp/init/default",
    "start:generate:help": "yarn run-cmd generate -h",
    "start:generate:model": "yarn run-cmd generate model -r ./.tmp/init/awesome",
    "start:g:model": "yarn run-cmd g model -r ./.tmp/init/awesome",
    "start:g:prisma": "yarn run-cmd g service -r ./.tmp/init/awesome prisma",
    "start:generate:run": "yarn run-cmd generate -r ./.tmp/init/awesome",
    "start:typeorm:run": "yarn run-cmd typeorm entity:create -r ./.tmp/init/awesome --name User",
    "test": "vitest run",
    "test:ci": "vitest run --coverage.thresholds.autoUpdate=true"
  },
  "bin": {
    "tsed": "lib/esm/bin/tsed.js"
  },
  "files": [
    "lib/esm/bin/tsed.js",
    "lib/esm/bin",
    "lib",
    "templates"
  ],
  "keywords": [
    "Ts.ED",
    "cli",
    "typescript",
    "framework",
    "rest",
    "api",
    "express",
    "decorators"
  ],
  "engines": {
    "node": ">=14"
  },
  "dependencies": {
    "@swc-node/register": "^1.10.9",
    "@swc/core": "1.7.26",
    "@swc/helpers": "^0.5.13",
    "@tsed/cli-core": "workspace:*",
    "@tsed/core": ">=8.0.0",
    "@tsed/di": ">=8.0.0",
    "@tsed/hooks": ">=8.0.0",
    "@tsed/logger": ">=7.0.1",
    "@tsed/openspec": ">=8.0.0",
    "@tsed/schema": ">=8.0.0",
    "chalk": "^5.3.0",
    "change-case": "^5.4.4",
    "esm-module-alias": "^2.2.1",
    "globby": "^14.0.2",
    "read-pkg-up": "^11.0.0",
    "semver": "^7.6.3",
    "tslib": "^2.7.0"
  },
  "devDependencies": {
    "@tsed/typescript": "workspace:*",
    "@types/change-case": "^2.3.1",
    "cross-env": "7.0.3",
    "typescript": "5.6.2",
    "vitest": "2.1.1"
  },
  "peerDependencies": {
    "@tsed/core": ">=8.0.0",
    "@tsed/di": ">=8.0.0",
    "@tsed/hooks": ">=8.0.0",
    "@tsed/logger": ">=7.0.1",
    "@tsed/openspec": ">=8.0.0",
    "@tsed/schema": ">=8.0.0"
  },
  "peerDependenciesMeta": {
    "@tsed/core": {
      "optional": false
    },
    "@tsed/di": {
      "optional": false
    },
    "@tsed/hooks": {
      "optional": false
    },
    "@tsed/logger": {
      "optional": false
    },
    "@tsed/openspec": {
      "optional": false
    },
    "@tsed/schema": {
      "optional": false
    }
  }
}
