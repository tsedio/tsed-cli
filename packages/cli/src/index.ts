import "./templates/index.js";

export * from "./Cli.js";
export * from "./commands/add/AddCmd.js";
export * from "./commands/generate/GenerateCmd.js";
export * from "./commands/init/config/FeaturesPrompt.js";
export * from "./commands/init/config/FeaturesPrompt.js";
export * from "./commands/init/InitCmd.js";
export * from "./commands/init/prompts/getFeaturesPrompt.js";
export * from "./commands/update/UpdateCmd.js";
export * from "./constants/index.js";
export * from "./fn/exec.js";
export * from "./fn/render.js";
export * from "./interfaces/index.js";
export * from "./pipes/index.js";
export * from "./runtimes/index.js";
export * from "./services/CliProjectService.js";
export * from "./services/CliTemplatesService.js";
export * from "./services/ProjectClient.js";
export * from "./utils/defineTemplate.js";
