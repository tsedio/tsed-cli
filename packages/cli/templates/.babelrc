{
  "presets": [
    ["@babel/env", {"targets": {"node": true}}],
    "@babel/typescript"
   ],
  "plugins": [
    "babel-plugin-transform-typescript-metadata",
    ["@babel/plugin-proposal-decorators", { "legacy": true }],
    ["@babel/plugin-proposal-class-properties", { "loose": true }],
    "@babel/proposal-object-rest-spread"
  ]
}