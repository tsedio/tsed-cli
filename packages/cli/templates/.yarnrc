"@tsedio:registry" "https://npm.pkg.github.com/tsedio"