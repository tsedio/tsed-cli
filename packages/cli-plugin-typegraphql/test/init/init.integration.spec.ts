import {CliPlatformTest, FakeCliFs} from "@tsed/cli-testing";
import {InitCmd, TEMPLATE_DIR} from "@tsed/cli";
import "@tsed/cli-plugin-typegraphql";

describe("Init TypeGraphQL project", () => {
  beforeEach(() =>
    CliPlatformTest.bootstrap({
      templateDir: TEMPLATE_DIR,
      commands: [InitCmd]
    })
  );
  afterEach(() => CliPlatformTest.reset());

  it("should generate a project with typegraphql", async () => {
    CliPlatformTest.setPackageJson({
      name: "",
      version: "1.0.0",
      description: "",
      scripts: {},
      dependencies: {},
      devDependencies: {}
    });

    await CliPlatformTest.exec("init", {
      platform: "express",
      convention: "conv_default",
      rootDir: "./project-data",
      projectName: "project-data",
      tsedVersion: "5.58.1",
      graphql: true
    });

    expect(FakeCliFs.getKeys()).toMatchInlineSnapshot(`
      Array [
        "./project-name",
        "project-name",
        "project-name/.barrelsby.json",
        "project-name/.dockerignore",
        "project-name/.gitignore",
        "project-name/Dockerfile",
        "project-name/README.md",
        "project-name/docker-compose.yml",
        "project-name/package.json",
        "project-name/processes.config.js",
        "project-name/src",
        "project-name/src/Server.ts",
        "project-name/src/config",
        "project-name/src/config/envs",
        "project-name/src/config/envs/index.ts",
        "project-name/src/config/index.ts",
        "project-name/src/config/logger",
        "project-name/src/config/logger/index.ts",
        "project-name/src/controllers/rest",
        "project-name/src/controllers/rest/HelloWorldController.ts",
        "project-name/src/datasources",
        "project-name/src/datasources/MyDataSource.ts",
        "project-name/src/datasources/index.ts",
        "project-name/src/index.ts",
        "project-name/src/resolvers",
        "project-name/src/resolvers/index.ts",
        "project-name/src/resolvers/recipes",
        "project-name/src/resolvers/recipes/Recipe.ts",
        "project-name/src/resolvers/recipes/RecipeNotFoundError.ts",
        "project-name/src/resolvers/recipes/RecipeResolver.ts",
        "project-name/src/services",
        "project-name/src/services/RecipeService.ts",
        "project-name/tsconfig.compile.json",
        "project-name/tsconfig.json",
      ]
    `);

    const content = FakeCliFs.entries.get("project-name/src/Server.ts")!;
    expect(content).toMatchSnapshot();
    expect(content).toContain("import \"@tsed/typegraphql\"");
    expect(content).toContain("import \"./datasources/index\";");
    expect(content).toContain("import \"./resolvers/index\";");

    const configContent = FakeCliFs.entries.get("project-name/src/config/index.ts")!;
    expect(configContent).toMatchSnapshot();
  });
});
