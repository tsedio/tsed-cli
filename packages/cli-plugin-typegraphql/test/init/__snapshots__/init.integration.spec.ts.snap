// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Init TypeGraphQL project should generate a project with typegraphql 2`] = `
"import {join} from \\"path\\";
import {Configuration, Inject} from \\"@tsed/di\\";
import {PlatformApplication} from \\"@tsed/common\\";
import \\"@tsed/platform-express\\"; // /!\\\\ keep this import
import \\"@tsed/ajv\\";
import \\"@tsed/typegraphql\\";
import \\"./datasources/index\\";
import \\"./resolvers/index\\";
import {config} from \\"./config/index\\";
import * as rest from \\"./controllers/rest/index\\";

@Configuration({
  ...config,
  acceptMimes: [\\"application/json\\"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  disableComponentsScan: true,
  ajv: {
    returnsCoercedValues: true
  },
  mount: {
    \\"/rest\\": [
      ...Object.values(rest)
    ]
  },
  middlewares: [
    \\"cors\\",
    \\"cookie-parser\\",
    \\"compression\\",
    \\"method-override\\",
    \\"json-parser\\",
    { use: \\"urlencoded-parser\\", options: { extended: true }}
  ],
  views: {
    root: join(process.cwd(), \\"../views\\"),
    extensions: {
      ejs: \\"ejs\\"
    }
  },
  exclude: [
    \\"**/*.spec.ts\\"
  ]
})
export class Server {
  @Inject()
  protected app: PlatformApplication;

  @Configuration()
  protected settings: Configuration;
}
"
`;

exports[`Init TypeGraphQL project should generate a project with typegraphql 3`] = `
"import {readFileSync} from \\"fs\\";
import {envs} from \\"./envs/index\\";
import loggerConfig from \\"./logger/index\\";
const pkg = JSON.parse(readFileSync(\\"./package.json\\", {encoding: \\"utf8\\"}));

export const config: Partial<TsED.Configuration> = {
  version: pkg.version,
  envs,
  logger: loggerConfig,
  graphql: {
    default: {
      path: \\"/graphql\\",
      buildSchemaOptions: {
      }
    }
  },
  // additional shared configuration
};
"
`;
